ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,580 @@
+                "\n        await workspace.Query<PartitionByReportingNodeAndPeriod>()",
+                "\n            .Where(x => x.Scenario == null)",
+                "\n            .OrderByDescending(x => x.Year)",
+                "\n            .ThenByDescending(x => x.Month)",
+                "\n            .Select(x => ParseReportingPeriodToDisplayString(x.Year, x.Month, separator))",
+                "\n            .Where(x => userInput == null || x.Contains(userInput, StringComparison.OrdinalIgnoreCase))",
+                "\n            .ToArrayAsync();",
+                "\n",
+                "\n    async void InitReportingPeriod() {",
+                "\n        ReportingPeriod = ParseReportingPeriodToDisplayString(GetStorage().InitialReportingPeriod.Year, GetStorage().InitialReportingPeriod.Month, separator);",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Filters"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface BasicFilterFormsEntity : MutableScopeWithWorkspace {",
+                "\n    [DropdownValues(\"\", \"GroupOfContract\", \"Novelty\", \"EconomicBasis\")]",
+                "\n    string FilterName { get; set; }",
+                "\n",
+                "\n    [DropdownMethod(nameof(GetBasicFilterAsync))]",
+                "\n    string FilterValue { get; set; }",
+                "\n",
+                "\n    [DropdownValues(\"\", \"Add\", \"Remove\")]",
+                "\n    string FilterAction { get; set; }",
+                "\n    ",
+                "\n    async Task<IReadOnlyCollection<string>> GetBasicFilterAsync(string userInput, int page, int pageSize) =>",
+                "\n        new string[]{ null }.Concat(",
+                "\n            await (FilterName switch",
+                "\n            {",
+                "\n                \"GroupOfContract\" => workspace.Query<DataNode>().Select(x => x.SystemName),",
+                "\n                \"Novelty\" => workspace.Query<Novelty>().Select(x => x.SystemName),",
+                "\n                \"EconomicBasis\" => workspace.Query<EconomicBasis>().Select(x => x.SystemName),",
+                "\n                _ => Enumerable.Empty<string>().AsQueryable()",
+                "\n            }) .Where(x => userInput == null || x.Contains(userInput, StringComparison.OrdinalIgnoreCase)).OrderBy(x => x).ToArrayAsync()).ToArray();",
+                "\n        ",
+                "\n    [NotVisible] IReadOnlyCollection<(string filterName, string filterValue)> InputDataFilter { get; set; }",
```

Comment:
Can we initialize it here as Enumerable.Empty<(string, string)>.ToArray() to avoid checking each time whether it is null or not

Reply:
Maybe... this was not working before, but after the bug on Scope inheritance was fixed, this could now indeed work as you suggest.

Reply:
postponed to the final version

Reply:
It works.